test/test_xnan_functions.cpp

```diff
@@ -227,6 +277,10 @@ namespace xt
         CHECK_RESULT_TYPE(nanvar(INPUT, {1, 2}), double);         \
         CHECK_RESULT_TYPE(nanstd(INPUT, {1, 2}), double);
 
+#define CHECK_NAN_MIN_MAX_RESULT_TYPE(INPUT, RESULT_TYPE)         \
+        CHECK_RESULT_TYPE(nanmin(INPUT, {1, 2}), RESULT_TYPE)     \
+        CHECK_RESULT_TYPE(nanmax(INPUT, {1, 2}), RESULT_TYPE)
```

Comment:
Test that the return type is not promoted automatically. I guess this happened in the previous version before the reducer refactoring.

I am not sure whether it is worth to also test the templated version.